File: 203-remove-linked-list-elements/203-remove-linked-list-elements.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* removeElements(struct ListNode* head, int val){
    
    if (head==NULL){
        
        return head;
    }
    
    struct ListNode* r_head=malloc(sizeof(struct ListNode));
    
    r_head->next=head;
    
    struct ListNode* sol=r_head;
    
    while (r_head->next != NULL){
        
        if (r_head->next->val==val){
            r_head->next=r_head->next->next;
        }
        
        else{
            
            r_head=r_head->next;

        }
        
    }
    
    return sol->next;
}

File: 203-remove-linked-list-elements/203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        
        if head==None:
            return head
        
        List=ListNode(0)
        List.next=head
        
        res=List
        
        while(List.next!=None):
            
            if List.next.val==val:
                
                List.next=List.next.next
            else:
                List=List.next
            
        return res.next
            
        